DOC-6703 explicitly parse SHOW BACKUP options

ashleykwtam opened the pull request:
Addresses DOC-6703:
- Added an example to show how a user can now pass and check multiple KMS URIs in SHOW BACKUP
- Updated the options parameter and note for locality aware incremental_locations

Preview:
- [Take & Restore Encrypted Backups](https://deploy-preview-17474--cockroachdb-docs.netlify.app/docs/stable/take-and-restore-encrypted-backups#show-a-backup-with-multi-region-encryption)
- [`SHOW BACKUP`](https://deploy-preview-17474--cockroachdb-docs.netlify.app/docs/v23.1/show-backup)

msbutler commented on src/current/v23.1/take-and-restore-encrypted-backups.md:
```diff
@@ -163,6 +163,45 @@ BACKUP INTO 's3://{BUCKET NAME}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY
 
+#### Show a backup with multi-region encryption
+
+To view a backup with [multi-region encryption](#multi-region), use the `kms` option and the same KMS URIs that were used to create the backup:
```
nit: this `#multi-region` link takes me to a somewhat random part of the docs page. Where did you intend it to go?

msbutler replied:
nit: s/backup with multi-region encryption/a kms encrypted locality aware backup/r 

I'd replace this elsewhere on this page too.

ashleykwtam commented:
via Slack - @msbutler wanted to confirm with the team when describing a locality aware backup with kms encryption, whether it is more appropriate to call it just that, “a locality aware backup with kms encryption”? Or, call it a “backup with multi region encryption”? 

@livlobo

livlobo commented:
I prefer “a locality aware backup with kms encryption”

MichaelTrestman reviewed (approved):
LGTM

msbutler commented:
fwiw: i see the phrase "multiregion encryption" elsewhere in the docs. It seems worth zapping those in this pr, but I'd be fine replacing them later, so long as there's an issue to track that.

ashleykwtam commented:
> fwiw: i see the phrase "multiregion encryption" elsewhere in the docs. It seems worth zapping those in this pr, but I'd be fine replacing them later, so long as there's an issue to track that.

removed instances of "multiregion encryption" and replaced with "locality-aware backup with kms encryption"
